onnxscript/version_converter/version_converter.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+from onnxscript import ir
+from onnxscript.version_converter._adapter_lib import pick_adapter_set
+
+
+class _VersionConverter:
+    def __init__(self, model: ir.Model) -> None:
+        self._functions = model.functions
+        self.model_version = model.opset_imports.get("")
+        self.custom_adapters = {}
+
+    def _register_custom_adapters(self, custom_adapter_list) -> None:
+        self.custom_adapters = custom_adapter_list
+
+    def graph_version_convert(self, graph: ir.Graph, target_version: int) -> None:
+        if self.model_version == target_version:
+            # No conversion needed
+            return
+
+        # Iterate through all nodes in the graph
+        # Set node.version as the model_version
+        # TODO: Do this internally in IR?
+        for node in graph:
+            node.version = self.model_version
+
+        # Iterate from current model version -> target version
+        # Updating each node based on the correct adapter [ver->ver+1]
+        for opset_version in range(self.model_version, target_version):
+            for node in graph:
+                adapter_set = pick_adapter_set(opset_version)
+                if node.op_type in adapter_set:
+                    adapter_func = adapter_set[node.op_type]
+                    if adapter_func is not None:
+                        adapter_func(node)
+                    node.version = opset_version + 1
+                # Apply custom adapters
```

Comment:
I would not bother with custom adapters to begin with. I don't see a need for it right now. But there are other features (will comment in other places as needed) that are far more important. It's better to not expose extra functionality for now (until we see a good reason to have it).

Reply:
Removed